inc/Classes/MasterForm.php

```diff
@@ -506,7 +506,11 @@ public function SendForm($BaseURL, $table, $idname = '', $id = 0)
                     $db_query = '';
                     if ($this->SQLFields) {
                         foreach ($this->SQLFields as $val) {
-                            $db_query .= ", $val";
+                            if ($SQLFieldTypes[$val] == 'varbinary(16)' and $val == 'ip') {
```

Comment:
Why is a comparison needed by the type (`varbinary`)? I have the feeling that this will lead to a bug later on once we change this because the code relies on the data field.

Reply:
Well, we already [have this in when adding/updating values](https://github.com/lansuite/lansuite/blob/36af57dbb00321b27895f98733f138be365d85ee/inc/Classes/MasterForm.php#L1122). This now fixes the same thing for retrieving values.
If we'd ever touch the fields we need to update both, however touching the fields would also mean we will most likely need to modify INET6_NTOA/ATON to something else anyway.
I think this is currently quite a clean approach because it applies these data modifications only if both field name and type match to what is required by these functions. Anything else and it is being dealt with like a regular database field without any modifications.
On future changes I'd assume that we'd either update both or add additional code for the necessary transformations which would also ensure that we'd stay compatible with older database layouts (e.g. the current one here).

Reply:
Any news? I think this is currently one of the very last major open bugs of 5.0-dev/Master and users will certainly hit it.

Reply:
Thanks for providing more context. Fine with me.

Reply:
Related to @M4LuZ comment:
For the LanSuite v5, there might be more issues. Yep. I remember there was a ticket of things we aligned on. Not sure where it is right now.

Related to the `varbinary(16)` comparison. I am aware that this is not beautiful. But it is working and consistent across the code (as stated by @aj-gh). I think this is more important right now (when i see the status of this project).

Reply:
@M4LuZ I am waiting for your response/action before merging this. I am fine by merging this. Don't want to do it when you disagree.

Reply:
> I'm sorry to disagree, there are still a good amount of issues left open, especially the upgrade handling from 4.2 -> 5.0
> That is also a place where the field conversion to `varbinary(16)` could be done to avoid this code here.

Even if you fully convert everything - you will still need either this or similar code unless you expect users to be able to parse IP addresses in varbinary format (as that is being shown to users when not doing INET6_NTOA on them, so without this code) ;)

Reply:
> I'm sorry to disagree, there are still a good amount of issues left open, especially the upgrade handling from 4.2 -> 5.0

Well, missing upgrade handling doesn't make core features (which I consider the `server` module to be a part of) on a freshly installed 5.0-dev/Master unuseable - that's what I would consider a major bug. And of these there shouldn't be that many left (see below) given that we run 5.0-dev/Master for our upcoming event.

> For the LanSuite v5, there might be more issues. Yep. I remember there was a ticket of things we aligned on. Not sure where it is right now.

[Here is the ToDo list](https://github.com/lansuite/lansuite/issues/224) (that would probably need another update) while [this is a list of issues and PRs tagged for 5.0-RC](https://github.com/lansuite/lansuite/milestones/LanSuite%205.0%20RC) - seems manageable.

Reply:
Push :)

Reply:
> Related to the `varbinary(16)` comparison. I am aware that this is not beautiful. But it is working and consistent across the code (as stated by @aj-gh). I think this is more important right now (when i see the status of this project).

^this, thus merging it while not being that happy about it ;)